pkg/networkservice/mechanisms/directmemif/server.go

```diff
@@ -80,11 +80,13 @@ func (d *directMemifServer) Request(ctx context.Context, request *networkservice
 		prev := d.proxies[conn.Id]
 		if prev != nil {
 			_ = prev.Stop()
+			d.executor.AsyncExec(func() {
```

Comment:
NOTE: we need to do this async exec inside another async exec in case of request. Because we need to be sure that deleting from line 73 has completed.

Reply:
This case also covered by unit test `TestServerReRequest`

Reply:
The good news is, you can do nested AsyncExecs :)